pkg/pipeline/pipeline.go

```diff
@@ -43,6 +45,9 @@ type Pipeline struct {
 	Extractor    extract.Extractor
 	Encoder      encode.Encoder
 	IsRunning    bool
+
+	start    []*node.Init
+	terminal []*node.Terminal
```

Comment:
@eranra @mariomac We should discuss the `Terminal` node (e.g. an export function) and whether to do away with the notions of `Transform`, `Extract`, etc that we currently have.